### PR TITLE
CHIA-685 Remove homebrew rpaths from _ssl.cpython.so on macOS during build

### PR DESCRIPTION
**Pull request by haorldbchi**
Removes rpath load commands from `_ssl.cpython-*.so` which are sometimes added on Apple M-series CPUs, prefer bundled dynamic libraries for which there is an rpath added already as `@loader_path/..` -- however, the homebrew rpaths appear with higher precedence, potentially causing issues.

See: #18099

### CODE REVIEW

**Comment by janit** [comment]
Thanks for this. Using bundled libs seems the most robust way for distributables.

Just a nitpick that using Brew is not related to CPU architecture. So also x86 Macs are just as likely to have packages sourced from there.

At a glance seems the script is universal for all macOS builds. So I imagine ARM and x86 both use the same script (just in a different environment) so both will be covered by this.

**Comment by hoffmang9** [comment]
Until the 2.3.1 release there were no external dependencies and this only shows up on ARM Macs as it's a change to either brew's paths on Mac ARM only or a change in sub components in Python tools.

**Review by altendky — CHANGES_REQUESTED** [review]
I would like to see a test for this, probably in:
https://github.com/Chia-Network/chia-blockchain/blob/a83b2401973d5e8709d897b0d8bd0fa3972b4333/chia/cmds/installers.py#L53

Also, hopefully we can identify where this issue is (pyinstaller?) and have at least a bug report with a minimal reproduction to reference in this workaround (I think it is anyways).

**Comment by emlowe** [comment]
There is an interesting combination of factors here.
When building python with `pyenv` it will add in Homebrew paths to the RPATH of the Python modules only when homebrew is not installed in `/usr/local/` as `/usr/local/lib` is in the default search path - on Apple Silicon, homebrew is installed in `/opt/homebrew` so the RPATHs are updated - this was changed in https://github.com/pyenv/pyenv/pull/2865

We recently (between 2.3.0 and 2.3.1) started to clean our build environment such that we always rebuild Python with pyenv each time we create the chia installer.  When 2.3.0 was released, we didn't rebuild python as the build system already had the version we wanted to use - this happened to have been built before the pyenv change above. However, with 2.3.1, we built Python with the pyenv version that performed that rpath manipulation.

On Intel Mac homebrew by default is in `/usr/local` therefore this issue doesn't impact the Intel build system and the packaged DMGs are ok on that platform

We aren't entirely sure if this is something pyinstaller should handle, but in the meantime we will workaround the issue by making sure to remove the offending RPATH from the library before final packaging

**Review by altendky — APPROVED** [review]

File: .github/workflows/build-macos-installers.yml [hunk 1]
```diff
@@ -450,6 +450,11 @@ jobs:
         run: |
           find "/Volumes/Chia "*
 
+      - name: Test for homebrew rpath
+        if: matrix.arch.name == 'ARM64'
```
Comment (altendky) [inline .github/workflows/build-macos-installers.yml:454]: This is what the `.matrix` elements are for.  This way when we bicker about what the name output should be we don't have to update all the references in the workflow.

```suggestion
        if: matrix.arch.matrix == 'arm'
```

Or, does running this in the Intel case cause trouble?

```suggestion
```

**Review by altendky — CHANGES_REQUESTED** [review]

File: .github/workflows/build-macos-installers.yml (hunk 1 above)
Comment (altendky) [reply .github/workflows/build-macos-installers.yml:454]: I'll propose my changes in another PR to keep this moving.

**Review by altendky — DISMISSED** [review]

**Review by arvidn — APPROVED** [review]

**Pull request merged**